documentation/projects/proposals/rekognition_data/20240530-implementation_plan_augment_catalog_with_rekognition_tags.md

```diff
@@ -0,0 +1,669 @@
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+In order to incorporate accomplish the goals of this plan, the following steps
+will need to be performed:
+
+1. [Determine which labels to exclude from Rekognition's label set](#determine-excluded-labels)
+2. [Remove the above excluded labels and tags below threshold from existing Clarifai tags](#filter-clarifai-tags)
+3. [Generate and insert the new Rekognition tags](#insert-new-rekognition-tags)
+
+## Step details
+
+<!--
+Describe all of the implementation steps listed in the "step-by-step plan" in detail.
+
+For each step description, ensure the heading includes an obvious reference to the step as described in the
+"step-by-step plan" section above.
+-->
+
+```{note}
+Some of the steps listed below have some cross-over with functionality defined
+in/required by the
+[data normalization project](/projects/proposals/data_normalization/20240227-implementation_plan_catalog_data_cleaning.md)
+(#430). Where possible, existing issues will be referenced and possible duplicated
+effort will be identified.
+```
+
+### Determine excluded labels
+
+This will involve a manual process of looking through each of the [available
+labels for Rekognition][aws_rekognition_labels] and seeing if they match any of
+the criteria to be filtered. The excluded labels should then be saved in an
+accessible location, either on S3 or within the
+[sensitive terms repository](https://github.com/WordPress/openverse-sensitive-terms)
+as a new file. Consent & approval should be sought from two other maintainers on
+the accuracy of the exclusion list prior to publishing.
+
+### Filter Clarifai tags
+
+```{attention}
+A snapshot of the catalog database should be created prior to running this step
+in production.
+```
+
+While this project seeks to add new magine-generated labels to the database, we
+already have
+[around 10 million records](https://github.com/WordPress/openverse/pull/3948#discussion_r1552301581)
+which include labels from the
+[Clarifai image labeling service](https://www.clarifai.com/products/scribe-data-labeling-platform).
+It is unclear how these labels were applied, or what the exhaustive label set
+is. Given how comprehensive Rekognition's label list is, I feel confident that
+the exclusions we identify from that list will be sufficient for filtering out
+unwanted demographic labels that Clarifai has used as well.
+
+Once the excluded labels are determined, we will need to filter those values
+from the existing Clarifai tags. The existing tags also include Clarifai labels
+that are below our [accuracy threshold](#accuracy-selection). These will be
+removed by the data normalization project (#430) and do not need to be
+considered at this time.
```

Comment:
I realized this as part of writing this proposal - we're going to be using the "cleaned tags TSV" saved from the data refresh process to update the tags in the catalog. Right now we're excluding tags below 0.9 confidence: https://github.com/WordPress/openverse/blob/3747f9aa40ed03899becb98ecae2abf926c8875f/ingestion_server/ingestion_server/cleanup.py#L133

This means that the tags below that confidence will be removed from the cleaned tags for a record, so when we update the catalog using that TSV, they'll also be removed.

Reply:
Which issue is that? I'd like to block it based on this discussion about how to use the catalog database.

Reply:
#3912

I certainly did not realize we'd be deleting data as part of this when we initially embarked on that project, and I agree that preserving it is the right move for this case.

Reply:
It was implicit in #3415 but we can exclude tags from there since we're shifting to use the `batched_update` DAG for that. Blocked while the discussion takes place ➕ 1️⃣